.travis.yml

```
@@ -1,13 +1,40 @@
 services:
 - docker
-script:
-- echo "Build the TODO list"
+
+branches:
+  only:
+    - master
+    - module_8
+
+
+before_script:
+- echo "Test the TODO list"
 - docker build --target testing-offline --tag todo-app:devops-course-starter_todoapp-test-offline .
 - docker run --env-file .env.test todo-app:devops-course-starter_todoapp-test-offline
 - docker build --target testing-online --tag todo-app:devops-course-starter_todoapp-test-online .
 - docker run  -e TRELLO_API_KEY=$TRELLO_API_KEY -e TRELLO_API_SECRET=$TRELLO_API_SECRET todo-app:devops-course-starter_todoapp-test-online
+
+      
+script: 
+- echo "Build the TODO list"
+- docker build --target production --tag $DOCKER_USER/todo-app:$TRAVIS_COMMIT .
+
+- echo "Push the TODO list to Docker Hub"
+- echo $DOCKER_PWD | docker login --username $DOCKER_USER --password-stdin
+- docker push $DOCKER_USER/todo-app:$TRAVIS_COMMIT
+
+- echo "Push the TODO list to Heroku"
+- docker pull $DOCKER_USER/todo-app:$TRAVIS_COMMIT
+- docker tag $DOCKER_USER/todo-app:$TRAVIS_COMMIT registry.heroku.com/georgestodoapp/web
+- HEROKU_API_KEY=$HEROKU_API_KEY heroku container:login
+- docker push registry.heroku.com/georgestodoapp/web
+- HEROKU_API_KEY=$HEROKU_API_KEY heroku container:release web -a georgestodoapp
```

You shouldn't the `HEROKU_API_KEY=$HEROKU_API_KEY` parts here (as `HEROKU_API_KEY` is an environment variable that will automatically be picked up by Heroku)
```suggestion
- heroku container:login
- docker push registry.heroku.com/georgestodoapp/web
- heroku container:release web -a georgestodoapp
```

---

Done